### PR TITLE
fix(exec): use full seq as slot identifier to prevent concurrent overflow collision

### LINKED ISSUE #1274: fix(exec): use unique slot identifier for interleaved overflow files
## Summary

Concurrency correctness issue found during the v0.22.3 audit (docs/audit/2026-07-05-v0.22.3.md, finding F10).

## Finding

**F10 -- LOW:** `crates/aptu-coder/src/tools/exec_command.rs` -- interleaved output overflow uses `slot = seq % 8`, mapping all concurrent requests to one of 8 fixed slot directories. The overflow file within each slot is named `interleaved` (not `interleaved-{seq}`). Two concurrent requests that hash to the same slot and both trigger interleaved overflow write to the same file path and silently overwrite each other's data. Under load (more than 8 concurrent `exec_command` calls with large interleaved output), this produces corrupted overflow data with no error surfaced.

## Fix

Replace `seq % 8` with the full `seq` value as the slot/file identifier, making each request's overflow file unique:

```rust
// before
let slot = seq % 8;
// after
let slot = seq; // unique per request
```

Add a regression test for concurrent overflow to `crates/aptu-coder/tests/exec_command.rs`.

## Files
- `crates/aptu-coder/src/tools/exec_command.rs`
- `crates/aptu-coder/tests/exec_command.rs`

### PR DESCRIPTION
**Pull request by clouatre**
## Summary

Replace `seq % 8` with `seq` at line 839 of exec_command.rs so each concurrent exec_command request gets a unique slot directory, eliminating the interleaved-overflow file collision that occurs when multiple concurrent requests share the same slot identifier.

## Changes

- crates/aptu-coder/src/tools/exec_command.rs
- crates/aptu-coder/tests/exec_command.rs

## Test plan

- [x] Tests pass (79 passed, 0 failed; see 03-build.json test_results)
- [x] Linter clean (cargo clippy -p aptu-coder -- -D warnings)
- [x] Security scan clean (no findings; see 04-validation.json security_summary)

## Details

The fix changes the slot calculation from `seq % 8` to `seq`, allowing each concurrent request to maintain a unique slot directory path. The modulo operation caused collisions when concurrent requests were processed, leading to interleaved overflow files being written to the same directory.

A new regression test `test_concurrent_interleaved_overflow` verifies that 4 concurrent exec_command calls over a single MCP connection each produce distinct interleaved_path values, preventing recurrence of this issue.

Closes #1274

### CODE REVIEW

File: crates/aptu-coder/src/tools/exec_command.rs [hunk 1]
```diff
@@ -836,7 +836,7 @@ pub(crate) async fn run_exec_impl(
         }
     }
 
-    let slot = seq % 8;
+    let slot = seq;
```
[the remark this answers is not in the document]
Comment (clouatre) [reply crates/aptu-coder/src/tools/exec_command.rs:839]: Acknowledged. Unbounded slot directory growth is a known side-effect documented in the guard analysis for this fix. It is tracked as a separate concern -- cleanup policy (session-scoped teardown or periodic retention) will be addressed in a follow-up. This PR is intentionally scoped to the collision fix only.
